pennylane/ops/op_math/sum.py

```diff
@@ -0,0 +1,156 @@
+
+    res = None
+    try:
+        for i, mat in enumerate(mats_gen):
+            res = mat if i == 0 else math.add(res, mat)
+    except MatrixUndefinedError as error:
+        print(f"\nThe matrix method must be defined for all summands: \n")
+        raise error
+
+    if dtype is not None:
+        res = math.cast(res, dtype)
+    if cast_like is not None:
+        res = math.cast_like(res, cast_like)
+
+    return res
+
+
+class Sum(Operator):
+    """Arithmetic operator subclass representing the sum of operators"""
+
+    _eigs = {}  # cache eigen vectors and values like in qml.Hermitian
+
+    def __init__(self, *summands, do_queue=True, id=None):
+
+        if len(summands) < 2:
+            raise ValueError(f"Require at least two operators to sum; got {len(summands)}")
+
+        self.summands = summands
+
+        combined_wires = qml.wires.Wires.all_wires([s.wires for s in summands])
+        combined_params = []
+        for s in summands:
+            combined_params += s.parameters
+
+        super().__init__(*combined_params, wires=combined_wires, do_queue=do_queue, id=id)
+        self._name = "Sum"
+
+    def __repr__(self):
+        """Constructor-call-like representation."""
+        return " + ".join([f"{f}" for f in self.summands])
+
+    def __copy__(self):
+        cls = self.__class__
+        copied_op = cls.__new__(cls)
+        copied_op.data = self.data.copy()  # copies the combined parameters
+        copied_op.summands = tuple(s.__copy__() for s in self.summands)
+
+        for attr, value in vars(self).items():
+            if attr not in {"data", "summands"}:
+                setattr(copied_op, attr, value)
+
+        return copied_op
+
+    @property
+    def num_wires(self):
+        return len(self.wires)
+
+    @property
+    def is_hermitian(self):
+        """If all of the terms in the sum are hermitian, then the Sum is hermitian."""
+        return all(s.is_hermitian for s in self.summands)
+
+    def terms(self):
+        return [1.0] * len(self.summands), self.summands
+
+    @property
+    def eigendecomposition(self):
+        """Return the eigendecomposition of the matrix specified by the Hermitian observable.
+
+        This method uses pre-stored eigenvalues for standard observables where
+        possible and stores the corresponding eigenvectors from the eigendecomposition.
+
+        It transforms the input operator according to the wires specified.
+
+        Returns:
+            dict[str, array]: dictionary containing the eigenvalues and the eigenvectors of the operator
+        """
+        Hmat = self.matrix()
+        Hmat = qml.math.to_numpy(Hmat)
+        Hkey = tuple(Hmat.flatten().tolist())
+        if Hkey not in self._eigs:
+            w, U = np.linalg.eigh(Hmat)
+            self._eigs[Hkey] = {"eigvec": U, "eigval": w}
+
+        return self._eigs[Hkey]
+
+    def diagonalizing_gates(self):
+        """Compute diagonalizing_gates (only if op is hermitian)"""
```

Comment:
Says "only if op is hermitian", but then has nothing to prevent non-hermitian ops from return values:
```
op = Sum(qml.RX(1.234, wires=0), qml.PauliZ(1))
op.diagonalizing_gates()
```
works without errors.

I don't think there's anything wrong with defining `diagonalizing_gates` for non-hermitian ops.  They just might not be useful in those situations.